### DISCUSSION TITLE
Several Arrays but select only any 2 to use

### DISCUSSION THREAD

**Discussion by HobmanX** [pschatzmann/arduino-audio-tools#1524] [Q&A] [answered]
If I have a sketch setup to generate sound from 2 Arrays and I want to be able to use different Arrays from time to time, but only select any 2 from a number that I would have uploaded into my project board. Basically select them on demand from the user controls.
The initialisation and sound generation within Audio Tools is all done before Setup in Arduino so is it possible to get back and rerun this selecting 2 different arrays etc and still use the same volume control, resampling etc already setup for the sketch...??

**Comment by pschatzmann** [comment, marked as answer]
Why don't you look up the name of the function to replace the array in the GeneratorFromArray or to replace the generator in the GeneratedSoundStream yourself ?

**Reply by HobmanX** [reply]
I assume you mean expressions like this...

void | setArray (T *array, size_t size)

void | setArray (T(&array)[arrayLen])

My problem is reading the documentation and understanding how to apply it, except I can see examples of typical use or explaination of how it is used rather than just stating it.

I tried various combinations but none worked....

void 	waveAR1.setArray(T(&array)[45]);     // at top of sketch to declare it..????

    waveAR1.setArray ( T(&AR45)[45]);     // in task2 to call it...
              waveAR1.begin();

........................

**Reply by pschatzmann** [reply]
It's just setArray(array)!

**Reply by HobmanX** [reply]
Yes, that works, Thanks....... But where do I get that information. Is it on Github. I browse through the Wiki and cannot find it anywhere.
The only references I found was the ones I posted last post and it was looking an &Array and an array length and a T that compiler told me was not in scope etc etc..
I cannot find setArray(array) anywhere......... and for it to work I had to put in the generatorname.setArray, so in my case waveAR1.setArray(AR1);   and    waveAR2.setArray(AR2);

**Comment by pschatzmann** [comment]
https://stackoverflow.com/questions/67260487/passing-an-array-to-function-without-explicit-array-size

**Reply by HobmanX** [reply]
Thanks for the info....but I would need to be reading the stages before that to understand the detail...when I was studying I tried to avoid pointers...???
and the question still remains  where do I find.........   setArray(array).......  in your documentation or a list of functions available...??

**Reply by pschatzmann** [reply]
I am sorry but I don't quite understand your question: you would just look at the available methods for the class in question using the class documentation

**Reply by HobmanX** [reply]
My question has been in several posts.....where do I get information to describe how to relate the items and commands in Audio tools which are used to make changes to the sketch and provide control.  In Arduino IDE I can go to the Reference Page in Help and see a list of all commands and functions that can be used and read about them and see examples of use. Where do I get similar list in Audio Tools.??
Is this link the type of page information you refer to..... https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_generator_from_array.html
This lists names of a lot of functions etc but I cannot find any description on how they relate to my sketch or how to use them,
As I said earlier.... setArray(array) as such is not listed. setArray is there in 2 places but with different parameters and I didn't know how to use it.
I also see setFrequency is there.......Does that mean I could use set frequency to give a particular frequency output instead of resampling and having to calculate the step size each time...??
